### PR TITLE
Fix wrong network detected for substrate address in explore mode

### LINKED ISSUE #4457: Wrong network detected in explore mode for substrate addresses
**Describe the bug**

Currently all addresses are detected as kusama ones, which is incorrect.

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS: 
 - Browser: 
 - Version: 

**Smartphone (please complete the following information):**
 - Device: 
 - OS: 
 - Browser: 
 - Version: 

**Additional context**

### PR DESCRIPTION
**Pull request by mariopino**
Closes #4457 

**Description:**

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by faboweb** [comment]
🚀 Deployed on https://5f04add4253fd2d17f249452--lunieio.netlify.app

File: app/src/vuex/modules/connection.js [hunk 1]
```diff
@@ -76,16 +76,19 @@ export default function ({ apollo }) {
...
           .filter(({ network_type }) => network_type === `polkadot`)
-          .find(
-            async (network) =>
-              await isValidPolkadotAddress(
-                address,
-                parseInt(network.address_prefix)
-              )[0]
-          )
-
+        for (let index = 0; index < substrateNetworks.length; index++) {
```
Comment (jbibla) [inline app/src/vuex/modules/connection.js:82]: why does the for loop work better than the filter?

Comment (mariopino) [reply app/src/vuex/modules/connection.js:82]: Filter and find are synchronous and doesn't work well with async code inside them. Same for forEach as I tested, but for loops works pretty well (https://stackoverflow.com/questions/43031988/javascript-efficiency-for-vs-foreach)

**Comment by Bitcoinera** [comment]
Not working completely... The url shows `polkadot-testnet` but I'm still in Kusama:

image



Or maybe I missed the scope of this PR?

**Comment by Bitcoinera** [comment]
Ok, it is working. And I didn't change anything. So weird

**Pull request merged**